alpine/src/main/scala/io/delta/alpine/internal/SnapshotImpl.scala

```diff
@@ -53,6 +56,13 @@ private[internal] class SnapshotImpl(
   override def getDeltaLog: DeltaLog = deltaLog
   override def getTimestamp: Long = timestamp
 
+  override def open(): CloseableIterator[RowParquetRecordJ] =
+    CloseableParquetDataIterator(
+      allFilesScala.map(_.path),
```

Comment:
We need to use this method to get the real file path: https://github.com/delta-io/delta/blob/master/src/main/scala/org/apache/spark/sql/delta/files/TahoeFileIndex.scala#L80 Could you also add tests for this?

Reply:
in my source code, what would be the correct `path` as used in here?
```
  protected def absolutePath(child: String): Path = {
    val p = new Path(new URI(child))
    if (p.isAbsolute) {
      p
    } else {
      new Path(path, p)
    }
  }
```

Reply:
spent a little while looking through use cases of the TahoeFileIndex, no answer really jumped out at me

Reply:
@scottsand-db It should be the data path.

Reply:
This will decode the string `%2D` to `-`. AddFile.path is an encoded string.

Reply:
thanks